### PR TITLE
[DOCFIX] Improve K8s doc

### PR DESCRIPTION
**Pull request by jiacheliu3**
This change:
1. Adds a bunch of notes on caveats
2. Removes redundant instruction on formatting journal job
3. Adds one note to Troubleshooting
4. Corrects a bunch of typos

### CODE REVIEW

File: docs/en/deploy/Running-Alluxio-On-Kubernetes.md [hunk 1]
```diff
@@ -448,14 +448,10 @@ Uninstall Alluxio as follows:
...
 $ kubectl delete -f ./worker/
 $ kubectl delete -f ./master/
-$ kubectl delete configmaps alluxio-config
-```
-
-(Optional) If using persistent volumes, execute the following to remove the persistent volume
-storing the Alluxio journal. Note: Alluxio metadata will be lost.
-```console
-$ kubectl delete -f alluxio-master-journal-pv.yaml
+$ kubectl delete configmap alluxio-config
```
Comment (jiacheliu3) [inline docs/en/deploy/Running-Alluxio-On-Kubernetes.md]: "configmaps" -> "configmap"

I'm sure the plural form will work tho.

File: docs/en/deploy/Running-Alluxio-On-Kubernetes.md [hunk 2]
```diff
@@ -448,14 +448,10 @@ Uninstall Alluxio as follows:
 ```console
 $ kubectl delete -f ./worker/
 $ kubectl delete -f ./master/
-$ kubectl delete configmaps alluxio-config
-```
-
-(Optional) If using persistent volumes, execute the following to remove the persistent volume
-storing the Alluxio journal. Note: Alluxio metadata will be lost.
-```console
-$ kubectl delete -f alluxio-master-journal-pv.yaml
```
Comment (jiacheliu3) [inline docs/en/deploy/Running-Alluxio-On-Kubernetes.md]: This is replaced by the Note.

File: docs/en/deploy/Running-Alluxio-On-Kubernetes.md [hunk 3]
```diff
@@ -504,11 +500,6 @@ Make sure all the Pods have been terminated before you move on to the next step.
 Check the Alluxio upgrade guide page for whether the Alluxio master journal has to be formatted.
 If no format is needed, you are ready to skip the rest of this section and move on to restart all Alluxio master and worker Pods.
 
-How the journal should be formatted depends on the Alluxio master [journal type]({{ '/en/operation/Journal.html#ufs-journal-vs-embedded-journal' | relativize_url }}).
-
-*UFS Journal*
-
-If you are running UFS journal, there is only one place for the journal.
```
Comment (jiacheliu3) [inline docs/en/deploy/Running-Alluxio-On-Kubernetes.md]: Now two Journal options will both use the job to format journal, so no need to have them separated.

File: docs/en/deploy/Running-Alluxio-On-Kubernetes.md [hunk 4]
```diff
@@ -643,6 +623,12 @@ Check access to the given port from a remote client using a network utility such
 $ nc -zv <IP> 29999
 ```
 
+### Permission Denied
+
+From Alluxio v2.1 on, Alluxio Docker containers except Fuse will run as non-root user `alluxio` with UID 1000 and GID 1000 by default.
+Kubernetes [`hostPath`](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath) volumes
+are only writable by root so you need to update the permission accordingly.
```
Comment (jiacheliu3) [inline docs/en/deploy/Running-Alluxio-On-Kubernetes.md]: Added one common gotcha since we changed the default user.

**Comment by madanadit** [comment]
alluxio-bot, merge this please.

**Comment by madanadit** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please.

**Pull request merged**